src/Form/Control/Multiline.php

```diff
@@ -324,24 +344,22 @@ public function saveRows(): self
         $currentIds = array_column($model->export(), $model->id_field);
 
         foreach ($this->rowData as $row) {
-            // should clone model to be able to save it multiple times
-            $row_model = clone $model;
+            $entity = $model->createEntity();
 
             foreach ($row as $fieldName => $value) {
                 if ($fieldName === '__atkml') {
                     continue;
                 }
 
-                if ($fieldName === $row_model->id_field && $value) {
-                    $row_model = $row_model->load($value);
+                if ($fieldName === $model->id_field && $value) {
+                    $entity = $model->load($value);
```

Comment:
Theoretically the id_field could be anywhere in the array, couldn't it? So it could be that some values are already set to the entity by the rows below ($entity->set($fieldName, $value), and then $entity->load($value) comes after. That could overwrite values already set by $entity->set($fieldName, $value).
I think load() should be the first thing happening, any set() should happen after.

Reply:
you are right - I guess it never breaks before probably because id is generally the first field in a row. I will make changes. Good catch!

Reply:
update